editor/src/messages/tool/tool_messages/artboard_tool.rs

```diff
@@ -133,7 +134,24 @@ impl ArtboardToolData {
 
 		Some(edges)
 	}
-
+	pub fn get_child_layers_of_selected_artboard(&self, document: &DocumentMessageHandler) -> Vec<LayerNodeIdentifier> {
+		let Some(artboard_id) = self.selected_artboard else {
+			return Vec::new();
+		};
+		let structure = document.metadata().structure.clone();
+		let child_layers = structure
+			.iter()
+			.filter_map(|(layer_id, node_relations)| {
+				if let Some(parent_id) = node_relations.parent {
+					if parent_id == artboard_id {
+						return Some(*layer_id);
+					}
+				}
+				None
+			})
+			.collect::<Vec<LayerNodeIdentifier>>();
+		child_layers
+	}
```

Comment:
You can use `self.selected_artboard.unwrap().children(&document.metadata())` to avoid this allocation.